tests/system/large/test_session.py

```diff
@@ -46,7 +50,72 @@
         # ),
     ],
 )
-def test_read_gbq_for_large_tables(session: Session, query_or_table, index_col):
+def test_read_gbq_for_large_tables(
+    session: bigframes.Session, query_or_table, index_col
+):
     """Verify read_gbq() is able to read large tables."""
     df = session.read_gbq(query_or_table, index_col=index_col)
     assert len(df.columns) != 0
+
+
+def test_close(session):
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    full_id_1 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    full_id_2 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_2)
+
+    session.close()
+
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_2)
+
+
+def test_clean_up_by_session_id():
+    original_location = bigframes.options.bigquery.location
+    bpd.close_session()  # close the session to change regions
+    bigframes.options.bigquery.location = "europe-west10"
+    # we do this test in a different region in order to avoid
+    # overly large amounts of temp tables slowing the test down
+
+    session = bpd.get_global_session()
+    session_id = session.session_id
+
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+    dataset = session._anonymous_dataset
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    tables_before = bqclient.list_tables(dataset, max_results=10000, page_size=10000)
+    tables_before_count = len(list(tables_before))
+    assert tables_before_count >= 2
+
+    bpd.clean_up_by_session_id(
+        session_id, location=session._location, project=session._project
+    )
+
+    tables_after = bqclient.list_tables(dataset, max_results=10000, page_size=10000)
+    assert len(list(tables_after)) <= tables_before_count - 2
+
+    bpd.close_session()  # close the session to change regions
+    bigframes.options.bigquery.location = original_location
```

Comment:
This can leave the tests in a weird state if the test fails. Let's use the [pytest monkeypatch fixture](https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.setattr) to ensure things are set to the previous value when the test finishes.

Probably should do this resetting via monkeypatch in the "global session" text fixture with the lock that I suggest above, actually.

Reply:
Assuming just using a new session object works, we can avoid the need for this entirely.